plutus-ledger-api/src/PlutusLedgerApi/Common/Eval.hs

```diff
@@ -250,9 +255,17 @@ evaluateScriptCounting ll pv verbose ectx p args = swap $ runWriter @LogOutput $
     let (res, UPLC.CountingSt final, logs) =
             evaluateTerm UPLC.counting pv verbose ectx appliedTerm
     tell logs
-    liftEither $ first CekError $ void res
+    case res of
+        Left e -> throwError (CekError e)
+        Right v -> unless (ll == PlutusV1 || ll == PlutusV2 || isUnit v) $ throwError NonUnitReturnValue
```

Comment:
Given that this logic is repeated twice and `tell logs` is repeated as well, I'd abstract it into a separate function like we did with `evaluateTerm`. I can imagine we'll want it for tests, new benchmarks etc, but even if not, I'd rather have it in a single place to make sure it never runs out of sync.

Reply:
Factored the common code out into `processLogsAndErrors`